lib/active_model/serializer.rb

```diff
@@ -86,6 +136,13 @@ def self.cache(options = {})
       self._cache_options = (options.empty?) ? nil : options
     end
 
+    # @param resource [ActiveRecord::Base, ActiveModelSerializers::Model]
+    # @return [ActiveModel::Serializer]
+    #   Preferentially retuns
+    #   1. resource.serializer
+    #   2. ArraySerializer for a collection
+    #   3. options[:serializer]
+    #   4. lookup serializer by class (i.e. resource is a class)
```

Comment:
`param resource [ActiveRecord::Base, ActiveModelSerializers::Model]` dunno how else to word it

Also, typo `returns`